src/main/java/com/autotune/analyzer/recommendations/engine/RecommendationEngine.java

```diff
@@ -2220,126 +2257,158 @@ public void fetchMetricsBasedOnProfileAndDatasource(KruizeObject kruizeObject, T
+                                            }
+
+                                            Method method = MetricAggregationInfoResults.class.getDeclaredMethod(KruizeConstants.APIMessages.SET + aggregationFunctionsEntry.getKey().substring(0, 1).toUpperCase() + aggregationFunctionsEntry.getKey().substring(1), Double.class);
+                                            method.invoke(metricAggregationInfoResults, value);
+                                            metricAggregationInfoResults.setFormat(format);
+                                            metricResults.setAggregationInfoResult(metricAggregationInfoResults);
+                                            metricResults.setName(metricEntry.getName());
+                                            metricResults.setFormat(format);
+                                            resMap.put(metricName, metricResults);
+                                            intervalResults.setMetricResultsMap(resMap);
+                                            intervalResults.setIntervalStartTime(sTime);  //Todo this will change
+                                            intervalResults.setIntervalEndTime(eTime);
+                                            intervalResults.setDurationInMinutes((double) ((eTime.getTime() - sTime.getTime())
+                                                    / ((long) KruizeConstants.TimeConv.NO_OF_SECONDS_PER_MINUTE
+                                                    * KruizeConstants.TimeConv.NO_OF_MSECS_IN_SEC)));
+                                            containerDataResults.put(eTime, intervalResults);
+                                            sTime = eTime;
                                         }
-                                    } catch (Exception e) {
-                                        throw new RuntimeException(e);
                                     }
+                                } catch (Exception e) {
+                                    throw new RuntimeException(e);
                                 }
                             }
-                            namespaceData.setResults(namespaceDataResults);
-                            if (!namespaceDataResults.isEmpty()) {
-                                setInterval_end_time(Collections.max(namespaceDataResults.keySet()));
-                            }
                         }
                     }
+
+                    containerData.setResults(containerDataResults);
+                    if (!containerDataResults.isEmpty())
+                        setInterval_end_time(Collections.max(containerDataResults.keySet()));    //TODO Temp fix invalid date is set if experiment having two container with different last seen date
+
                 }
             }
         } catch (Exception e) {
             e.printStackTrace();
             throw new Exception(AnalyzerErrorConstants.APIErrors.UpdateRecommendationsAPI.METRIC_EXCEPTION + e.getMessage());
         }
     }
+
```

Comment:
merge this fucntion into one

```
private String getMaxDateQuery(PerformanceProfile metricProfile, String metricName) {
    List<Metric> metrics = metricProfile.getSloInfo().getFunctionVariables();
    for (Metric metric: metrics) {
        String name = metric.getName();
        if(name.equals(metricName)) {
            return metric.getAggregationFunctionsMap().get("max").getQuery();
        }
    }
    return null;
}
```


`String containerMaxDateQuery = getMaxDateQuery(metricProfile, "maxDate");
String namespaceMaxDateQuery = getMaxDateQuery(metricProfile, "namespaceMaxDate");
`

Reply:
and see if we can rename "maxdate" to "conainterMaxDate" and please use constants

Reply:
Updated the code to use the common function as suggested.

Reply:
Changing "maxdate" to "conainterMaxDate" will require changes in constants, performance profiles and in JSON also. This may break few things, I haven't done this yet. I have noted this and will change everywhere.